config/backend-config/backend-config.go

```diff
@@ -308,3 +308,10 @@ func getNotOKError(respBody []byte, statusCode int) error {
 	}
 	return fmt.Errorf("backend config request failed with %d%s", statusCode, errMsg)
 }
+
+func (d *DestinationT) GetRouterIdentifier(isolateFlagMap map[string]bool) string {
```

Comment:
instead of passing the map as an argument, can't we have this method maintain a thread-safe global var map?

Reply:
I wanted to avoid introducing newer mutexes ... do you prefer having a global var over this?

Reply:
I admit I don't like global variables, but here for some reason I prefer it over the existing solution :) Global variable is used as a singleton config state and we only need to initialise these config variables once

Reply:
Modified it to use global var

Reply:
I don't think a global variable is a good idea for caching configs. It makes testing with different parameters tricky and can be hard to debug. 

I believe these smells point to the fact that we need a different abstraction to for making this mapping